x-pack/plugins/cloud/public/plugin.ts

```diff
@@ -198,7 +204,22 @@ export class CloudPlugin implements Plugin<CloudSetup> {
       if (userId) {
         // Do the hashing here to keep it at clear as possible in our source code that we do not send literal user IDs
         const hashedId = sha256(userId.toString());
-        fullStory.identify(hashedId);
+        const kibanaVer = this.initializerContext.env.packageInfo.version.split('.');
+        application?.then(async () => {
+          const appStart = await application;
+          appStart.currentAppId$.subscribe((appId) => {
+            // Update the current application every time it changes
+            fullStory.setUserVars({
+              appId: appId ?? 'unknown',
+            });
+          });
+        });
+        fullStory.identify(hashedId, {
+          version_str: this.initializerContext.env.packageInfo.version,
+          version_major_int: kibanaVer[0] || -1,
+          version_minor_int: kibanaVer[1] || -1,
+          version_patch_int: kibanaVer[2] || -1,
```

Comment:
I think we can use `semver.parse` instead of manually splitting the version string.